thirty-three-across/thirty-three-across-htb-system-tests.js

```diff
@@ -56,6 +56,10 @@ function validateBidRequest(request) {
         w: sizes[1][0],
         h: sizes[1][1]
     });
+    expect(queryObj.imp[0].banner.ext).toBeDefined();
+    expect(queryObj.imp[0].banner.ext.ttx).toBeDefined();
+    expect(queryObj.imp[0].banner.ext.ttx.viewability).toBeDefined();
+    expect(queryObj.imp[0].banner.ext.ttx.viewability.amount).toBeDefined();
```

Comment:
Can we not test this more by mocking element dimensions etc anc checking for actual values in different scenarios?

Reply:
we can not really do this, one is these viewability measurement are heavily depends on the DOM, and mocking the DOM is not easy. second is this test case is not control by us, we can not run this test case for different scenarios

Reply:
Hmm in Prebid we tested like so https://github.com/33Across/Prebid.js/blob/34cd13384033c5ee8d54d552c621f463325e6696/test/spec/modules/33acrossBidAdapter_spec.js#L146

Not asking you to mock the DOM. Asking you to mock the dimensions returned by your helper functions.